From 5b427134d613a49bcbd2fe89e7cac938d664b9e7 Mon Sep 17 00:00:00 2001
From: David Kupka <dkupka@redhat.com>
Date: Tue, 9 Aug 2016 17:03:25 +0200
Subject: [PATCH] frontent: Add summary class property to CommandOverride

Avoid creating instance of overriden command to get its summary.

https://fedorahosted.org/freeipa/ticket/6048

Reviewed-By: Jan Cholasta <jcholast@redhat.com>
---
 ipaclient/frontend.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ipaclient/frontend.py b/ipaclient/frontend.py
index aeaed550771d3c6af04a9b34fcae414faacb47d7..587e31c89b3935984e799f7d4c500c652bcb5d43 100644
--- a/ipaclient/frontend.py
+++ b/ipaclient/frontend.py
@@ -127,6 +127,12 @@ class CommandOverride(Command):
     doc = classproperty(__doc_getter)
 
     @classmethod
+    def __summary_getter(cls):
+        return cls.__get_next().summary
+
+    summary = classproperty(__summary_getter)
+
+    @classmethod
     def __NO_CLI_getter(cls):
         return cls.__get_next().NO_CLI
 
-- 
2.7.4

